src/gcp_scanner/scanner.py

```diff
@@ -274,8 +300,11 @@ def crawl_loop(initial_sa_tuples: List[Tuple[str, Credentials, List[str]]],
 
       sa_results_data = json.dumps(sa_results, indent=2, sort_keys=False)
 
-      with open(out_dir + '/%s.json' % project_id, 'a',
-                encoding='utf-8') as outfile:
+      output_file_name = generate_unique_filename(f'{project_id}-{current_time}'
+                                                  , '.json', out_dir)
+      output_file_path = os.path.join(out_dir, output_file_name)
+
+      with open(output_file_path, 'a', encoding='utf-8') as outfile:
```

Comment:
Same feedback as for the other PR: change to mode "x" to avoid race conditions and move the file creation to early in the execution.